src/metabase/lib/order_by.cljc

```diff
@@ -0,0 +1,74 @@
+(ns metabase.lib.order-by
+  (:require
+   [clojure.spec.alpha :as s]
+   [metabase.lib.dispatch :as lib.dispatch]
+   [metabase.lib.field :as lib.field]
+   [metabase.lib.options :as lib.options]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.schema.order-by :as lib.schema.order-by]
+   [metabase.lib.util :as lib.util]
+   [metabase.util.malli :as mu]))
+
+(defmulti ^:private ->order-by-clause
+  {:arglists '([query stage-number x])}
+  (fn [_query _stage-number x]
+    (lib.dispatch/dispatch-value x)))
+
+(defmethod ->order-by-clause :asc
+  [_query _stage-number clause]
+  (lib.options/ensure-uuid clause))
+
+(defmethod ->order-by-clause :desc
+  [_query _stage-number clause]
+  (lib.options/ensure-uuid clause))
+
+;;; by default, try to convert `x` to a Field clause and then order by `:asc`
```

Comment:
Not historical.

in terms of REPL and test usage, if you were to do something like 

```clj
(-> (lib/query (mt/id :venues))
    (lib/order-by "ID"))
```

it seemed like treating `"ID"` as a Field would be more obvious than generating SQL like

```sql
SELECT * 
FROM venues
ORDER BY 'ID' ASC
```

(The same goes for integers -- makes more sense to treat them as Field IDs rather than doing something like `ORDER BY 1` or whatever I think)

But we can change it if you disagree.